src/main/java/org/cbioportal/web/columnar/BasicDataBinner.java

```diff
@@ -0,0 +1,306 @@
+package org.cbioportal.web.columnar;
+
+import org.cbioportal.model.*;
+import org.cbioportal.service.StudyViewColumnarService;
+import org.cbioportal.web.columnar.util.NewClinicalDataBinUtil;
+import org.cbioportal.web.parameter.*;
+import org.cbioportal.web.util.DataBinner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Component;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.Collections.emptyList;
+
+@Component
+public class BasicDataBinner {
```

Comment:
Make some kind of comment this should eventually deprecate ClinicalDataBinner. Since this is a copy of that class but generalized

Reply:
comments added